challenge/dns01/precheck.go

```diff
@@ -23,23 +24,47 @@ func WrapPreCheck(wrap WrapPreCheckFunc) ChallengeOption {
 	}
 }
 
+// DisableCompletePropagationRequirement obsolete.
+// Deprecated: use DisableAuthoritativeNssPropagationRequirement instead.
 func DisableCompletePropagationRequirement() ChallengeOption {
+	return DisableAuthoritativeNssPropagationRequirement()
+}
+
+func DisableAuthoritativeNssPropagationRequirement() ChallengeOption {
+	return func(chlg *Challenge) error {
+		chlg.preCheck.requireAuthoritativeNssPropagation = false
+		return nil
+	}
+}
+
+func RecursiveNSsPropagationRequirement() ChallengeOption {
 	return func(chlg *Challenge) error {
-		chlg.preCheck.requireCompletePropagation = false
+		chlg.preCheck.requireAuthoritativeNssPropagation = true
```

Comment:
```suggestion
		chlg.preCheck.requireRecursiveNssPropagation = true
```
Calling `RecursiveNSsPropagationRequirement()`should set `chlg.preCheck.requireRecursiveNssPropagation`. Not authoritative.

Reply:
good catch